src/h5web/vis-packs/core/containers/ScalarVisContainer.tsx

```diff
@@ -15,6 +16,7 @@ function ScalarVisContainer(props: VisContainerProps): ReactElement {
   assertBaseType(entity);
 
   const value = useDatasetValue(entity);
+  assertNumOrStrOrBool(value);
   return <ScalarVis value={value} />;
```

Comment:
This is where it gets tricky.

The result of `useDatasetValue` is `unknown` and `ScalarVis` now expects `number | string | boolean`. According to TS, these types are incompatible as `unknown` is not assignable to `true`. 

I therefore moved the `assert` from the Vis in the Container.

Reply:
This is because of my new super-fancy magical typing of `useDatasetValue`. It returns `unknown` because the `HDF5BooleanType` needs to be added to the conditional type we use there. Once that's done, you won't need `assertNumOrStrOrBool`.